test/buffer.jl

```diff
@@ -83,6 +84,21 @@ using FilePathsBase: FileBuffer
         end
     end
 
+    @testset "eof" begin
+        mktempdir(SystemPath) do d
+            p, _ = mktemp(d)
+            io = FileBuffer(p; write=true)
+            @test eof(io)
```

Comment:
i'm not sure if this new testset is worth it, but we didn't have a test for `eof` when passed an empty file, so i though i'd add that, and to be explicit about what (i think) the expected interaction is with read and write